source/nodejs/adaptivecards-designer/src/card-designer.ts

```diff
@@ -431,8 +431,38 @@ export class CardDesigner extends Designer.DesignContext {
         this.recreateDesignerSurface();
 
         if (this._deviceEmulationChoicePicker) {
-            this._deviceEmulationChoicePicker.isEnabled = !!this.hostContainer.enableDeviceEmulation
-            this.activeDeviceEmulationChanged()
+            this._deviceEmulationChoicePicker.isEnabled = !!this.hostContainer.enableDeviceEmulation;
+            this.activeDeviceEmulationChanged();
+        }
+
+        if (this._containerSizeChoicePicker) {
+            this._containerSizeChoicePicker.isEnabled = !!this.hostContainer.supportsMultipleSizes;
+
+            // Update the host parameter data with the value of the choice picker
+            if (this._containerSizeChoicePicker.isEnabled && this._sampleHostData) {
```

Comment:
Personally, I think we are running into `bad smell`. Each time, we are adding a new control into the toolbar(deviceemulation first, then size and theme) for any container, then it adds everything to the base host-container class even other containers doesn't need it.
Is it possible to reverse the dependencies? For example, have a function:
hostContainer.UpdateToolBar() instead of 
 `if (hostContainer.supportsMultipleSizes....`. 

In this way, we don't need to keep `this._containerSizeChoicePicker.isEnabled` and it's just an internal state of berlin container.
Also doesn't need to make change on host-container most of time.

Reply:
I'm not sure if this is possible because the toolbar is not contained within the `Container` classes. There's not much that the container itself "owns" unless we want to do a major refactor.

@paulcam206 Do you have any ideas here?

Reply:
@licanhua I've filed a ticket for this here (https://github.com/microsoft/AdaptiveCards/issues/7773) so that we can refactor all of the host container classes at the same time.